File: cont/base/springcontent/shaders/GLSL/Particles/BubbleParticleGenerator.lua
return {
	MainCode =
[[
	vec4 partColor = vec4(alpha);
	
	float interSize = size + sizeExpansion * frameInfo.y;

	AddEffectsQuadCamera(
		drawOrder,
		vec3(1.0),
		pos, vec2(interSize), texCoord,
		partColor
	);
]]
}

File: rts/Rendering/Env/Particles/Generators/BubbleParticleGenerator.cpp
#include "BubbleParticleGenerator.h"

CR_BIND(BubbleParticleData, )
CR_REG_METADATA(BubbleParticleData,
(
	CR_MEMBER(pos),
	CR_MEMBER(alpha),
	CR_MEMBER(size),
	CR_MEMBER(sizeExpansion),
	CR_MEMBER(drawOrder),
	CR_MEMBER(unused),
	CR_MEMBER(texCoord)
))